README.md
# WILDNIGHT

# Intro

This is a night survival mod. 
Here, every night you will experience the dangers of biters. 
Victory can only be achieved by launching rockets. 

Rocket Rush mode is recommended. If playing in regular mode, the difficulty of the biters may be too high. 

In RocketRush mode, the gameplay duration is about 5 hours.

Gameplay Mix Cut: https://youtu.be/WLqRpNRN5zE
intro video(Chinese): https://www.youtube.com/watch?v=NqDU1JMjA-8

## Main Function

Once you enter the initial room of RocketRush, you will be able to convert 10,000 gold into an initial resource presets via the quick button in the upper left corner. 

The playtime is displayed in the upper left corner. Entering the night shows the current number of nights.

 The in-game daytime is doubled, and the vision in night is darker.

The in-game peace time is 45 minutes, and only a small number of biters will attack.

You need to use this time to build a defensive base in 45 minutes.
After that, biters attack each night, and as the number of days of play increases, that attack increases. 
You'll need to defend, gather resources, and launch rockets to achieve ultimate victory.

This module also has a series of balance tweaks to K2 and RampantFixed to make the gameplay more smooth. 

## Balancing Tweaks

Battle Robots: The price is cheaper. Lasts longer, has higher health, doesn't move now, and can be seen as a cheaper and more convenient defense tower. 

The robot duration is 10 minutes/15 minutes/30 minutes.

Gun turret:

The firing range of the machine-gun turret was slightly increased,
The number of stacks of ammunition in the machine gun turret was reduced, from 200 to 50, avoiding waste.
At the same time, the number of attacks per magazine has increased by 50%

Laser tower:

Slightly increased the range of the laser tower,Reduced power consumption of the laser tower. 

Others:

Increased resistance of the wall 
Increased resistance to building robots, which can now ignore biters attacks

 The manufacturing time of the main material of the rocket is reduced, avoiding long waiting times in the later stage

 Increase the water products of the air condenser in K2, from 30 to 120, which can be used in water-scarce environments

Removed the JuggernautEnemy in RampantFixed
Removed underground attacks in RampantFixed
poison enemies, fire enemies, and nuclear enemies in RampantFixed are turned off by default

## Options
Single Option to turn on the poison enemies, fire enemies, and nuclear enemies.
Should change the actual RampantFixed Option
Hard mode: biters have more health and attacks more often


## Others
Special thanks to Dimm2101(RampantFixed) Mod Creator for creating the AI remote interface for this mod

---------------------

## CHINESE

## 模组介绍

这是一个夜晚生存模组。
在这里，每一个夜晚你都会体验到异星虫群的危险。
只有通过建造防御基地，发射火箭才能获取胜利。


推荐使用Rocket Rush(即进击的火箭) 模式进行游戏。
如果在常规模式下游戏，虫群的难度可能太高。

在RocketRush模式下， 游戏时长约为5小时。

剪辑: https://youtu.be/WLqRpNRN5zE
简介: https://www.youtube.com/watch?v=NqDU1JMjA-8

## 模组主要功能
进入RocketRush的初始房间后，将可以通过左上角快速按钮将10000金币转换为一套初始资源配置，
节约选择时间。


左上角会显示游戏时间。进入夜晚会显示当前是第几个夜晚。

游戏内一天的时间扩大为两倍，夜晚的视野会更加昏暗
游戏内和平时间为45分钟，45分钟以内只会有很少的虫群进行攻击。
你需要利用这段时间初步建设好防御阵地

45分钟后，每个夜晚都会有虫群进行攻击，随着游戏天数的增加，虫群的数量也会不断增加。

你需要做好防御，采集资源，发射火箭才能获得最终胜利。

同时本模组对K2和RampantFixed还做了一系列平衡性改动使游戏性更加平滑。

## 主要平衡性改动
战斗机器人：
价格更加便宜。
持续时间更长，生命值更高，现在不会移动，可以看成更便宜更方便的防御塔。
机器人持续时间为分别为10分钟/15分钟/30分钟

机枪塔
略微增加了机枪塔的射程 
降低了机枪塔的弹药的堆叠数量，从200降低到50,避免过多的弹药浪费
同时每个弹匣的攻击次数增加了50%

激光塔
略微增加了激光塔的射程
降低了激光塔的耗电量。

其他
增加了墙体的抗性
增加了建造机器人的抗性，建造机器人现在可以无视虫群的攻击
降低了火箭的主要材料的制造时间，避免后期的长时间等待
增加K2中蒸汽冷凝器的水产量，从30提高到120,在缺水的环境中可以使用
去掉了RampantFixed中的主宰虫
去掉了RampantFixed中的地底攻击
默认关闭了RampantFixed中的毒液虫，火焰虫，核爆虫

## 模组可选项
毒液虫，火焰虫，核爆虫的开启选项。(需要更改RampantFixed的选项)
高难选项：该模式下，虫群的进攻会更频繁

依赖模组：
Krastorio2
RampantFixed

其他可选模组：
Jetpack： 喷气背包，方便行动
Larger Lamps: 光照范围更大的灯光


changelog.txt
---------------------------------------------------------------------------------------------------
Version: 0.5.6
Date: 2023-01-29
  Changes:
    - Decrease Night Attack Wave Strength in later game
---------------------------------------------------------------------------------------------------
Version: 0.5.5
Date: 2023-01-29
  Changes:
    - Increase Enemy HP and Night Attack Wave
---------------------------------------------------------------------------------------------------
Version: 0.5.4
Date: 2023-03-09
  Bugfixes:
    - change AI difficulty to Lite to make easier gameplay
---------------------------------------------------------------------------------------------------
Version: 0.5.2
Date: 2023-03-08
  Bugfixes:
    - Fix a potential bug if working with mutliple surface mods
---------------------------------------------------------------------------------------------------
Version: 0.5.1
Date: 2023-03-08
  Features:
    - Added text of current mode after victory


prototypes/battle.lua
------------------------------------------------
-- combat robot
data.raw["combat-robot"]["defender"]["time_to_live"]=60*60*10
data.raw["combat-robot"]["defender"]["max_health"] = 150
data.raw["combat-robot"]["defender"]["follows_player"]=false
data.raw["combat-robot"]["defender"]["light"]= {intensity = 0.3, size = 12, color={r=1, g=0.9}}
data.raw["combat-robot"]["defender"]["speed"] = 0

data.raw["combat-robot"]["distractor"]["max_health"] = 200
data.raw["combat-robot"]["distractor"]["time_to_live"]=60*60*15
data.raw["combat-robot"]["distractor"]["light"]= {intensity = 0.2, size = 14, color={r=1, g=0.9, b=0.5}}

data.raw["combat-robot"]["destroyer"]["max_health"]= 300
data.raw["combat-robot"]["destroyer"]["time_to_live"]=60*60*30
data.raw["combat-robot"]["destroyer"]["light"]= {intensity = 0.2, size = 16, color={r=1, g=0.9, b=0.8}}
data.raw["combat-robot"]["destroyer"]["follows_player"]=false
data.raw["combat-robot"]["destroyer"]["speed"] = 0
data.raw["combat-robot"]["distractor"].attack_parameters.range = 25
data.raw["combat-robot"]["destroyer"].attack_parameters.range = 30

local ingredients_table_magazine =
{
	{"coal", 1},
	{"iron-plate", 1},
	{"copper-plate", 1},
}

data.raw["recipe"]["rifle-magazine"]["ingredients"]= ingredients_table_magazine

local ingredients_table_defender =
{
	{"rifle-magazine", 2},
	{"iron-gear-wheel", 2},
    {"copper-cable", 2}
}

data.raw["recipe"]["defender-capsule"]["ingredients"]= ingredients_table_defender
data.raw["recipe"]["defender-capsule"]["energy_required"] = 5

local ingredients_table_distractor =
{
    {"glass", 2},
    {"electronic-circuit", 2},
    {"defender-capsule", 2},
}

data.raw["recipe"]["distractor-capsule"]["ingredients"]= ingredients_table_distractor
data.raw["recipe"]["distractor-capsule"]["energy_required"] = 5

local ingredients_table_destroyerr =
{
    {"electronic-components", 2},
    {"advanced-circuit", 2},
    {"distractor-capsule", 2},
}

data.raw["recipe"]["destroyer-capsule"]["ingredients"]= ingredients_table_destroyerr
data.raw["recipe"]["distractor-capsule"]["energy_required"] = 5


data.raw["projectile"]["defender-capsule"].action.action_delivery.target_effects =  {
  entity_name = "defender",
  show_in_tooltip = true,
  offsets = {
    {
      0.3,
      0.3
    },
    {
      -0.3,
      -0.3
    }
  },
  type = "create-entity"
}
data.raw["projectile"]["destroyer-capsule"].action.action_delivery.target_effects =  {
  entity_name = "destroyer",
  show_in_tooltip = true,
  offsets = {
    {
      0.4,
      -0.4
    },
    {
      -0.4,
      -0.4
    },
    {
      -0.4,
      0.4
    },
    {
      0.4,
      0.4
    },
  },
  type = "create-entity"
}



------------------------------------------------
-- railgun
data.raw["recipe"]["kr-railgun-turret"]["ingredients"]= {
    {"rare-metals", 20},
    {"steel-beam", 10},
    {"processing-unit", 4},
    {"low-density-structure", 4},
    {"gun-turret", 4},
}
data.raw["recipe"]["basic-railgun-shell"]["ingredients"]= {
    {"steel-plate", 4},
    {"electronic-circuit", 1},
}
data.raw["recipe"]["basic-railgun-shell"]["energy_required"]= 4


data.raw["recipe"]["explosion-railgun-shell"]["ingredients"]= {
    {"explosives", 1},
    {"basic-railgun-shell", 1},
}
data.raw["recipe"]["explosion-railgun-shell"]["energy_required"]= 4


data.raw["recipe"]["kr-rocket-turret"]["ingredients"]= {
    {"rare-metals", 20},
    {"steel-beam", 10},
    {"processing-unit", 10},
    {"low-density-structure", 4},
    {"rocket-launcher", 8},
}
data.raw["recipe"]["explosive-turret-rocket"]["ingredients"]= {
    {"steel-plate", 4},
    {"explosives", 2},
    {"electronic-circuit", 2},
}
data.raw["recipe"]["explosive-turret-rocket"]["energy_required"]= 6

------------------------------------------------
data.raw["recipe"]["battery-equipment"]["energy_required"]= 4
data.raw["recipe"]["big-battery-equipment"]["energy_required"]= 6
data.raw["recipe"]["big-battery-mk2-equipment"]["energy_required"]= 12
data.raw["recipe"]["big-battery-mk3-equipment"]["energy_required"]= 12
------------------------------------------------

-- data.raw["ammo-turret"]["gun-turret"].inventory_size = 50
--


data.raw["electric-turret"]["laser-turret"].max_health = 750
data.raw["electric-turret"]["laser-turret"].call_for_help_radius = 40
data.raw["electric-turret"]["laser-turret"].energy_source = {
  type = "electric",
  buffer_capacity = "1000kJ",
  input_flow_limit = "2000kW",
  drain = "50kW",
  usage_priority = "primary-input",
}

data.raw["electric-turret"]["laser-turret"].attack_parameters = {
  type = "beam",
  cooldown = 30,
  range = 40,
  source_direction_count = 64,
  source_offset = { 0, -3.423489 / 4 },
  damage_modifier = 3,
  ammo_type = {
    category = "laser",
    energy_consumption = "625kJ",
    action = {
      type = "direct",
      action_delivery = {
        type = "beam",
        beam = "laser-beam",
        max_length = 40,
        duration = 30,
        source_offset = { 0, -1.31439 },
      },
    },
  },
}

------------------------------------------------
-- robot
for _, robot in pairs(data.raw["construction-robot"]) do

    robot.resistances = {}
    for damageType, _ in pairs(data.raw["damage-type"]) do
        robot.resistances[damageType] = {
            type = damageType,
            percent = 100
        }
    end
end


data.raw["ammo"]["firearm-magazine"].stack_size=50
data.raw["ammo"]["firearm-magazine"].magazine_size=45

data.raw["ammo"]["rifle-magazine"].stack_size=50
data.raw["ammo"]["rifle-magazine"].magazine_size=45

data.raw["ammo"]["armor-piercing-rifle-magazine"].stack_size=50
data.raw["ammo"]["armor-piercing-rifle-magazine"].magazine_size=45


data.raw["ammo"]["uranium-rifle-magazine"].stack_size=50
data.raw["ammo"]["uranium-rifle-magazine"].magazine_size=45

data.raw["ammo"]["imersite-rifle-magazine"].stack_size=50
data.raw["ammo"]["imersite-rifle-magazine"].magazine_size=45

data.raw["gun"]["anti-material-rifle"].attack_parameters.range=80
